dto/api_input.py

```diff
@@ -55,50 +54,80 @@ def validate_password(self, data, **kwargs):
                 raise ValidationError("Invalid password format", field_name="password_hash")
         return data
 
+    @pre_load
+    def process_models(self, data, **kwargs):
+        if "models" in data and isinstance(data["models"], list):
+            processed_models = []
+            for item in data["models"]:
+                dto_class = self.detect_dto(item)
+                if dto_class:
+                    processed_models.append(dto_class().load(item))
+            data["models"] = processed_models
+        return data
+
+    def detect_dto(self, item):
+        type_map = {
+            "news": NewsDTO,
+            "league": TeamsLeagueDTO,
+            "game": GamesDTO
+        }
+        return type_map.get(item.get("type"))
 
 class TeamsLeagueDTO(BaseDTO):
     sport_id = fields.Int(required=False, missing=None)
     league_id = fields.Int(required=False, missing=None)
     country_id = fields.Int(required=False, missing=None)
-    letter = fields.Str(required=False, missing="")
     name = fields.Str(required=False, missing=None)
-    page = fields.Int(required=False, missing=0)
-    per_page = fields.Int(required=False, missing=0)
+
 
 class TeamsStatisticsOrPlayersDTO(BaseDTO):
     sport_id = fields.Int(required=False, missing=None)
+    sport = fields.Str(required=False, missing=None)
     team_id = fields.Int(required=False, missing=None)
     league_id = fields.Int(required=False, missing=None)
+    name = fields.Str(required=False, missing=None)
 
 
-class SearchDTO(BaseDTO):
-    leagues__sport_id = fields.Int(required=False, missing=None)
-    countries__country_id = fields.Int(required=False, missing=None)
-    letter = fields.Str(required=False, missing="")
+class FilterDTO(BaseDTO):
```

Comment:
Let's say we want to add 20 new filters. That'd force us to extend this class each time.

Plus, let's say I decided to order_by_desc my news. Do I need date_to, time_to etc etc if I'm not using them? That doesn't sound good.

I would like you to think of a way to make these filters more generic with the least amount of useless info/fields possible. 

What if we were like:

I want to sort `this` (e.g. games) by `this field` (e.g. number of views) in `this order` (e.g. ASC/DESC)

That'd allow you to create a general SearchEngine which would only require one API endpoint, basically